jpa/subsystem/src/main/java/org/jboss/as/jpa/config/PersistenceUnitMetadataImpl.java

```diff
@@ -379,6 +381,11 @@ public boolean needsJPADelegatingClassFileTransformer() {
 
     @Override
     public void addTransformer(ClassTransformer classTransformer) {
+        // Do not add Hibernate ORM 6.x/7.x bytecode enhancement class transformers
+        // (e.g. org.hibernate.jpa.internal.enhance.EnhancingClassTransformerImpl)
+        if(classTransformer.getClass().getName().startsWith(ORG_HIBERNATE_ORM_PROVIDER_CLASS_ENHANCER_PACKAGE)) {
+            return;
+        }
```

Comment:
Why do we not want to do this?

Reply:
Good question, I'm now thinking the comment should answer that.

Reply:
Pushed a comment change with the answer:
> WFLY-19694 Do not add Hibernate ORM 6.x/7.x bytecode enhancement class transformers (e.g. org.hibernate.jpa.internal.enhance.EnhancingClassTransformerImpl) which were already added earlier via org.jboss.as.jpa.hibernate.WildFlyClassTransformer or org.wildfly.persistence.jipijapa.hibernate7.WildFlyClassTransformer

Reply:
Will mark this resolved by ^ comment change.